go/store/prolly/tree/node_cursor.go

```diff
@@ -240,6 +241,82 @@ func NewLeafCursorAtItem(ctx context.Context, ns NodeStore, nd Node, item Item,
 	return cur, nil
 }
 
+type LeafSpan struct {
+	Leaves     []Node
+	LocalStart int
+	LocalStop  int
+}
+
+func FetchLeafNodeSpan(ctx context.Context, ns NodeStore, root Node, start, stop uint64) (LeafSpan, error) {
+	leaves, localStart, err := fetchLeafNodeSpan(ctx, ns, []Node{root}, start, stop)
+	if err != nil {
+		return LeafSpan{}, err
+	}
+
+	localStop := (stop - start) + localStart
+	for i := 0; i < len(leaves)-1; i++ {
+		localStop -= uint64(leaves[i].Count())
+	}
+
+	return LeafSpan{
+		Leaves:     leaves,
+		LocalStart: int(localStart),
+		LocalStop:  int(localStop),
+	}, nil
+}
+
+func fetchLeafNodeSpan(ctx context.Context, ns NodeStore, nodes []Node, start, stop uint64) ([]Node, uint64, error) {
```

Comment:
this will BFS and return the final recursive gets, all leaf nodes? might be helpful in the doc comment, took me a couple minutes to figure out. would it be useful at some point to have leaf pointers in the node cache or somewhere to avoid reading all intermediate nodes?

Reply:
yes, it does a breadth-first traversal of the ordinal range [start, stop). The BFS allows us to do faster batch reads via `NodeStore.ReadMany()`. I'll add a doc comment.

The intermediate nodes are the leaf pointers. AKAIK there's not a general way to store leaf pointers for this traversal that is more efficient than storing the internal nodes themselves